core/src/main/java/org/bitcoinj/core/BlockLocator.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2018.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.bitcoinj.core;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class BlockLocator {
+    List<Sha256Hash> blockLocator;
```

Comment:
Shouldn't this be named something like `hashes`? `blockLocator` seems misleading given that you just factored it out into a class.

Maybe make it private and use a getter? Consider making a BlockLocator immutable.